<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Gui</class>
 <widget class="QWidget" name="Gui">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Travelling Salesman Problem</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <!-- Верхняя часть: Поле ввода городов и кнопки -->
   <item>
    <layout class="QHBoxLayout" name="topLayout">
     <item>
      <widget class="QTextEdit" name="textEdit">
       <property name="placeholderText">
        <string>Введите координаты городов (x,y,1)...</string>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QVBoxLayout" name="buttonLayout">
       <item>
        <widget class="QPushButton" name="loadFileButton">
         <property name="text">
          <string>Загрузить из файла</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="generateRandomButton">
         <property name="text">
          <string>Сгенерировать случайно</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="inputCitiesButton">
         <property name="text">
          <string>Ввести города</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <!-- Параметры алгоритма -->
   <item>
    <layout class="QHBoxLayout" name="paramsLayout">
     <item>
      <widget class="QLabel" name="popSizeLabel">
       <property name="text">
        <string>Размер популяции:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="popSizeSpinBox">
       <property name="minimum">
        <number>10</number>
       </property>
       <property name="maximum">
        <number>1000</number>
       </property>
       <property name="value">
        <number>100</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="mutationRateLabel">
       <property name="text">
        <string>Вероятность мутации:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDoubleSpinBox" name="mutationRateSpinBox">
       <property name="decimals">
        <number>2</number>
       </property>
       <property name="minimum">
        <double>0.000000000000000</double>
       </property>
       <property name="maximum">
        <double>1.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.010000000000000</double>
       </property>
       <property name="value">
        <double>0.200000000000000</double>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="maxGenLabel">
       <property name="text">
        <string>Количество поколений:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="maxGenSpinBox">
       <property name="minimum">
        <number>10</number>
       </property>
       <property name="maximum">
        <number>1000</number>
       </property>
       <property name="value">
        <number>100</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="applyButton">
       <property name="text">
        <string>Применить</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <!-- Управление шагами -->
   <item>
    <layout class="QHBoxLayout" name="controlLayout">
     <item>
      <widget class="QPushButton" name="runStepButton">
       <property name="text">
        <string>Следующий шаг</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="backButton">
       <property name="text">
        <string>Предыдущий шаг</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="runToEndButton">
       <property name="text">
        <string>Запуск до конца</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="compareButton">
       <property name="text">
        <string>Сравнить решения</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="solutionComboBox"/>
     </item>
    </layout>
   </item>
   <!-- Информация о поколении -->
   <item>
    <layout class="QHBoxLayout" name="infoLayout">
     <item>
      <widget class="QLabel" name="generationLabel">
       <property name="text">
        <string>Поколение: 0</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="bestFitnessLabel">
       <property name="text">
        <string>Лучшая приспособленность: 0.00</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="avgFitnessLabel">
       <property name="text">
        <string>Средняя приспособленность: 0.00</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <!-- Графическое отображение и график -->
   <item>
    <layout class="QHBoxLayout" name="viewLayout">
     <item>
      <widget class="QGraphicsView" name="graphicsView">
       <property name="minimumSize">
        <size>
         <width>400</width>
         <height>400</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>400</width>
         <height>400</height>
        </size>
       </property>
       <property name="optimizationFlags">
        <set>QGraphicsView::DontSavePainterState</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QGraphicsView" name="compareGraphicsView">
       <property name="minimumSize">
        <size>
         <width>400</width>
         <height>400</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>400</width>
         <height>400</height>
        </size>
       </property>
       <property name="optimizationFlags">
        <set>QGraphicsView::DontSavePainterState</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QChartView" name="chartView">
     <property name="minimumSize">
      <size>
       <width>800</width>
       <height>400</height>
      </size>
     </property>
     <property name="maximumHeight">
      <number>400</number>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <!-- Подключения сигналов и слотов -->
  <connection>
   <sender>loadFileButton</sender>
   <signal>clicked()</signal>
   <receiver>Gui</receiver>
   <slot>on_loadFileButton_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>900</x>
     <y>50</y>
    </hint>
    <hint type="destinationlabel">
     <x>600</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>generateRandomButton</sender>
   <signal>clicked()</signal>
   <receiver>Gui</receiver>
   <slot>on_generateRandomButton_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>900</x>
     <y>100</y>
    </hint>
    <hint type="destinationlabel">
     <x>600</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>inputCitiesButton</sender>
   <signal>clicked()</signal>
   <receiver>Gui</receiver>
   <slot>on_inputCitiesButton_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>900</x>
     <y>150</y>
    </hint>
    <hint type="destinationlabel">
     <x>600</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>applyButton</sender>
   <signal>clicked()</signal>
   <receiver>Gui</receiver>
   <slot>on_applyButton_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>700</x>
     <y>200</y>
    </hint>
    <hint type="destinationlabel">
     <x>600</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>runStepButton</sender>
   <signal>clicked()</signal>
   <receiver>Gui</receiver>
   <slot>on_runStepButton_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>100</x>
     <y>250</y>
    </hint>
    <hint type="destinationlabel">
     <x>600</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>backButton</sender>
   <signal>clicked()</signal>
   <receiver>Gui</receiver>
   <slot>on_backButton_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>200</x>
     <y>250</y>
    </hint>
    <hint type="destinationlabel">
     <x>600</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>runToEndButton</sender>
   <signal>clicked()</signal>
   <receiver>Gui</receiver>
   <slot>on_runToEndButton_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>300</x>
     <y>250</y>
    </hint>
    <hint type="destinationlabel">
     <x>600</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>compareButton</sender>
   <signal>clicked()</signal>
   <receiver>Gui</receiver>
   <slot>on_compareButton_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>400</x>
     <y>250</y>
    </hint>
    <hint type="destinationlabel">
     <x>600</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>solutionComboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>Gui</receiver>
   <slot>on_solutionComboBox_changed(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>500</x>
     <y>250</y>
    </hint>
    <hint type="destinationlabel">
     <x>600</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>